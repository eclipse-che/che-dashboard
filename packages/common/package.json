{
  "name": "@eclipse-che/common",
  "version": "7.34.0-SNAPSHOT",
  "repository": "https://github.com/eclipse-che/che-dashboard",
  "license": "EPL-2.0",
  "private": true,
  "main": "lib/index.js",
  "engines": {
    "yarn": ">=1.20.0",
    "node": ">=12"
  },
  "scripts": {
    "build": "yarn clean && tsc -b",
    "watch": "yarn build && tsc -w",
    "clean": "rimraf lib tsconfig.tsbuildinfo"
  },
  "devDependencies": {
    "axios": "^0.21.1",
    "rimraf": "^3.0.2",
    "typescript": "~4.4.2"
  }
}
