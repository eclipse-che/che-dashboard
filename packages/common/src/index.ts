/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import helpers from './helpers';
import * as api from './dto/api';

export * from './dto/cluster-info';
export * from './dto/cluster-config';

export { helpers, api };

export const FACTORY_LINK_ATTR = 'factoryLink';
export const ERROR_CODE_ATTR = 'error_code';
export const KUBECONFIG_MOUNT_PATH = '/tmp/.kube';

const common = {
  helpers,
  api,
};
export default common;
