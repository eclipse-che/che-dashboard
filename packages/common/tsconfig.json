{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "composite": true
  },
  "include": [
    "src",
  ],
  "exclude": [
    "node_modules",
    "lib",
  ]
}
