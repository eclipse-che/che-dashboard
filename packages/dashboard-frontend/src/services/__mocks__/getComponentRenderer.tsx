/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

export default function <T extends Array<unknown>>(
  getComponent: (..._args: T) => React.ReactElement,
): {
  createSnapshot: (...args: T) => renderer.ReactTestRenderer;
  renderComponent: (...args: T) => {
    reRenderComponent: (..._args: T) => void;
  };
} {
  return {
    createSnapshot: (...args: T) => {
      return renderer.create(getComponent(...args));
    },
    renderComponent: (...args: T) => {
      const res = render(getComponent(...args));
      return {
        reRenderComponent: (..._args: T) => {
          res.rerender(getComponent(..._args));
        },
      };
    },
  };
}
