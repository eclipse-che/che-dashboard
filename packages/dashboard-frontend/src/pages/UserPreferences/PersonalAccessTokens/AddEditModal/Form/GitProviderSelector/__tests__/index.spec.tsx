/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { api } from '@eclipse-che/common';
import { Form } from '@patternfly/react-core';
import userEvent from '@testing-library/user-event';
import React from 'react';

import getComponentRenderer, { screen } from '@/services/__mocks__/getComponentRenderer';

import { GitProviderSelector } from '..';

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

const mockOnSelect = jest.fn();

describe('Registry Username Input', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const snapshot = createSnapshot();
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('default selected provider', () => {
    renderComponent();

    // the default dropdown value should be 'GitHub'
    const dropdownButton = screen.queryByRole('button', { name: 'GitHub' });
    expect(dropdownButton).toBeTruthy();
  });

  test('available provider options', async () => {
    renderComponent();

    // the default dropdown value should be 'GitHub'
    const dropdownButton = screen.getByRole('button', { name: 'GitHub' });
    await userEvent.click(dropdownButton);

    expect(screen.queryByRole('menuitem', { name: 'Bitbucket Server' })).toBeTruthy();
    expect(screen.queryByRole('menuitem', { name: 'GitHub' })).toBeTruthy();
    expect(screen.queryByRole('menuitem', { name: 'GitLab' })).toBeTruthy();
    expect(screen.queryByRole('menuitem', { name: 'Microsoft Azure DevOps' })).toBeTruthy();
  });

  it('should select a provider', async () => {
    renderComponent();

    // the default dropdown value should be 'GitHub'
    const dropdownButton = screen.getByRole('button', { name: 'GitHub' });
    await userEvent.click(dropdownButton);

    const bitbucketMenuitem = screen.getByRole('menuitem', { name: 'Bitbucket Server' });

    await userEvent.click(bitbucketMenuitem);

    expect(mockOnSelect).toHaveBeenCalledWith('bitbucket-server');
    expect(screen.queryByRole('button', { name: 'Bitbucket Server' })).toBeTruthy();
  });

  it('should handle component update', () => {
    const { reRenderComponent } = renderComponent('bitbucket-server');

    // expect Bitbucket to be selected by default
    expect(screen.queryByRole('button', { name: 'Bitbucket Server' })).toBeTruthy();

    reRenderComponent('gitlab');

    // expect the dropdown button to be updated
    expect(screen.queryByRole('button', { name: 'GitLab' })).toBeTruthy();
  });
});

function getComponent(provider?: api.GitProvider): React.ReactElement {
  return (
    <Form>
      <GitProviderSelector provider={provider} onSelect={mockOnSelect} />
    </Form>
  );
}
