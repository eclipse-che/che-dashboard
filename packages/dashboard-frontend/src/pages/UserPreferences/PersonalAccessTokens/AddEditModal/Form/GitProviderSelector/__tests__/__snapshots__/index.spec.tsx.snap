// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registry Username Input snapshot 1`] = `
<form
  className="pf-c-form"
  noValidate={true}
>
  <div
    className="pf-c-form__group"
  >
    <div
      className="pf-c-form__group-label"
    >
      <label
        className="pf-c-form__label"
        htmlFor="provider-name"
      >
        <span
          className="pf-c-form__label-text"
        >
          Provider Name
        </span>
      </label>
       
    </div>
    <div
      className="pf-c-form__group-control"
    >
      <div
        className="pf-c-dropdown"
        data-ouia-component-id="OUIA-Generated-Dropdown-1"
        data-ouia-component-type="PF4/Dropdown"
        data-ouia-safe={true}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="pf-c-dropdown__toggle"
          data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
          data-ouia-component-type="PF4/DropdownToggle"
          data-ouia-safe={true}
          disabled={false}
          id="toggle-initial-selection"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            className="pf-c-dropdown__toggle-text"
          >
            GitHub
          </span>
          <span
            className="pf-c-dropdown__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 320 512"
              width="1em"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</form>
`;
